src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -1234,6 +1234,17 @@ AS
         THROW 50404, 'Instance not found', 1;
     END
 
+    -- Deleting tag errors
+    DECLARE @wm BIGINT
```

Comment:
There can be multiple rows in `@deletedInstances` because the caller can delete a study or series. So you can simply `DELETE` all of the watermarks found in `@deletedInstances`